### PR TITLE
Fix L2ListenTcpdump tests (windows)

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- fixes the often failing TCPdump test on **windows**
- fixes OSX crashing on longer term https://github.com/secdev/scapy/pull/970
- may be used as an easy workaround in https://github.com/secdev/scapy/issues/870

### CODE REVIEW

**Comment by gpotter2** [comment]
@p-l- I haven't followed the entire https://github.com/secdev/scapy/issues/870 issue, but if the goal is to send and recieve packets on different sockets, this should be easily achievable with this PR:
see
```
read_s = L2ListenTcpdump()
out_s = conf.L2socket()
icmp_r = Ether()/IP(dst="secdev.org")/ICMP()
res = sndrcv(out_s, icmp_r, timeout=5, rcv_pks=read_s)
read_s.close()
out_s.close()

response = res[0][0][1]
assert response[ICMP].type == 0
```

**Comment by p-l-** [comment]
@gpotter2 can you commit only the tcpdump / windows part? For Travis / pip / OSX the issue is more complex and I'm trying to handle it.

**Comment by gpotter2** [comment]
@p-l- Sure thing ! Updated

**Comment by gpotter2** [comment]
@p-l- @guedou Should be mergeable even though the OSX tests failed

**Comment by p-l-** [comment]
@guedou could you rebase against master so that the OSX tests have a chance to pass?

**Comment by p-l-** [comment]
Sorry, I mean @gpotter2 could you rebase against master so that the OSX tests have a chance to pass?

**Comment by gpotter2** [comment]
@p-l- I've done it.
It seems that OSX as sudo only does not work with the new tests for other reasons :/
Should I just disable the tests for OSX ?

File: scapy/supersocket.py [hunk 1]
```diff
@@ -189,6 +189,8 @@ class L2ListenTcpdump(SuperSocket):
 
     def __init__(self, iface=None, promisc=None, filter=None, nofilter=False,
                  prog=None, *arg, **karg):
+        if DARWIN:
+            raise OSError("L2ListenTcpdump is not available on OSX because tcpdump does not support piped input.")
```
Comment (gpotter2) [inline scapy/supersocket.py]: For more info, see
https://github.com/secdev/scapy/blob/master/scapy/utils.py#L1276-L1292

**Comment by gpotter2** [comment]
@p-l- Rebased

File: scapy/supersocket.py (hunk 1 above)
Comment (p-l-) [reply scapy/supersocket.py]: I don't understand. This should be handled under Darwin by the code you are linking, shouldn't it?

Comment (gpotter2) [reply scapy/supersocket.py]: Well you're right :o just don't understand why this very test fails on OSX only :/ (the rest is working, i've rebased)

**Comment by gpotter2** [comment]
@p-l- Alright, fixed it. It was as on windows: `-i` is compulsory as auto-selection is broken

**Pull request merged**